command/agent/agent.go

```diff
@@ -244,6 +244,24 @@ func (a *Agent) consulConfig() *consul.Config {
 			Port: a.config.Ports.Server,
 		}
 	}
+	if a.config.AdvertiseAddrs.SerfLan != "" {
+		addr, err := net.ResolveTCPAddr("tcp", a.config.AdvertiseAddrs.SerfLan)
+		if err != nil {
+			fmt.Printf("Failed to parse AdvertiseAddrs.SerfLan: %s\n", err)
+		} else {
+			base.SerfLANConfig.MemberlistConfig.AdvertiseAddr = addr.IP.String()
+			base.SerfLANConfig.MemberlistConfig.AdvertisePort = addr.Port
+		}
+	}
+	if a.config.AdvertiseAddrs.SerfWan != "" {
+		addr, err := net.ResolveTCPAddr("tcp", a.config.AdvertiseAddrs.SerfWan)
+		if err != nil {
+			fmt.Printf("Failed to parse AdvertiseAddrs.SerfWan: %s\n", err)
+		} else {
+			base.SerfWANConfig.MemberlistConfig.AdvertiseAddr = addr.IP.String()
+			base.SerfWANConfig.MemberlistConfig.AdvertisePort = addr.Port
+		}
+	}
```

Comment:
Instead of printing a message for both of the above and allowing Consul to start even with a bad advertise address specified in its config, can we store the `*net.TCPAddr`s in the AdvertiseAddrs struct and parse the address while we are decoding? This would also make a config merge pretty easy for these fields. We do a similar thing with parsing time values.